### DISCUSSION TITLE
Can I use tinymce inside iframe with attribute sandbox="allow-scripts"?

### DISCUSSION THREAD

**Discussion by emiljas** [tinymce/tinymce#9348] [General]
I would like to use tinymce (react.js, self-hosted) inside custom PowerBI visual. It means my app needs to work properly inside iframe with attribute sandbox="allow-scripts".

It looks like that:
```
  <Editor
        onInit={(evt, editor) => editorRef.current = editor}
        initialValue="<p>This is the initial content of the editor.</p>"
        init={{
            skin: false,
            content_style: [contentCss, contentUiCss"].join('\n'),
            height: 500,
            menubar: false,
            plugins,
            toolbar
        }}
    />
```

Unfortunately I gets error:
>Uncaught DOMException: Failed to read a named property 'document' from 'Window': Blocked a frame with origin "null" from accessing a cross-origin frame.

I know tinymce creating other iframe and problem is with accessing parent's document/window for security reasons. Can I do something about that or tinymce is not option for my use case?

**Comment by lostkeys** [comment]
Have you tried inline mode? It doesn't isolate the content in  an iframe?

**Reply by emiljas** [reply]
@lostkeys Yes, inline mode works fine. Unfortunately I need editor with more classic look.

Out of curiosity, is there some reason for TinyMCE to not use cross window communication?

**Reply by TheSpyder** [reply]
@emiljas the document object is critical to editor functionality. We can't load a script into the content iframe and ping messages across the boundary to operate the editor; we need direct access to the document object of the iframe the editor creates.

We have speculated about creating a classic-looking editor with inline mode benefits using a div and shadow dom instead of an iframe, which sounds like it would work for you, but I believe last time we checked the state of browser selection for shadow dom meant it's not a viable user experience. Hopefully one day it can be.

**Reply by lostkeys** [reply]
@emiljas You can get a more classic look while still using inline mode with a [fixed toolbar container](https://www.tiny.cloud/docs/tinymce/latest/menus-configuration-options/#fixed_toolbar_container) and [toolbar persist](https://www.tiny.cloud/docs/tinymce/latest/menus-configuration-options/#toolbar_persist) config options. You won't get a statusbar though...

**Reply by emiljas** [reply]
@lostkeys  After some tests I think this suits my needs. Thank you for help.
